keras/src/ops/core_test.py

```diff
@@ -1345,3 +1347,59 @@ def test_associative_scan_invalid_arguments(self):
             core.associative_scan(
                 lambda x, y: (x[0] + y[0], x[1] + y[1]), x, axis=1
             )
+
+
+class CoreOpsRematTest(testing.TestCase):
+    def test_remat_basic_call(self):
+        if backend.backend() in ("openvino", "numpy"):
+            self.skipTest(
+                "remat is not supported in openvino and numpy backends."
+            )
+
+        def build_and_train_model(
+            use_remat, x_train, y_train, epochs, batch_size
+        ):
+            if use_remat:
+
+                def relu_fn(x):
+                    return activations.relu(x)  # Explicitly pass x
+
+                intermediate_function = backend.core.remat(relu_fn)
+            else:
+                intermediate_function = (
+                    activations.relu
+                )  # Just use relu normally
+
+            inputs = layers.Input(shape=(4,))
+            x = layers.Dense(4)(inputs)
+            x = layers.Lambda(lambda x: intermediate_function(x))(x)
+            outputs = layers.Dense(1)(x)
+            model = models.Model(inputs=inputs, outputs=outputs)
+            model.predict(x_train)
+            model.compile(optimizer="sgd", loss="mse")
+
+            # Train model
+            start_time = time.time()
+            model.fit(
+                x_train,
+                y_train,
+                epochs=epochs,
+                batch_size=batch_size,
+                verbose=0,
+            )
+            end_time = time.time()
+
+            # Calculate time and memory
+            time_taken = end_time - start_time
+            return time_taken
```

Comment:
This is not used. Also you're not calculating memory.

Just remove?

I don't know if there is a easy way to check if it's saving memory.

Reply:
I had tried to add a memory test to check total memory during training. But, for a smaller model it does not make a difference and also total memory used would give flaky results because we will need to check only the memory increment for model.fit().